WordPress/src/main/java/org/wordpress/android/ui/reader/services/update/ReaderUpdateLogic.java

```diff
@@ -321,7 +322,13 @@ private void handleFollowedBlogsResponse(final JSONObject jsonObject) {
             public void run() {
                 ReaderBlogList serverBlogs = ReaderBlogList.fromJson(jsonObject);
                 ReaderBlogList localBlogs = ReaderBlogTable.getFollowedBlogs();
-
+                // Remove duplicates from the server blogs list only if local and remote  lists don't match.
+                if (serverBlogs.size() != localBlogs.size()) {
```

Comment:
❓ Wouldn't it make sense to always remove the duplicates, independent of the lists sizes?

I understand that checking the size is a small optimization since the `isSameList` method below also compares the list sizes but removing the duplicates looks like something we always want to do to avoid incorrect logic (e.g.: right now our local table doesn't contain duplicates just because we use only the `blogId` as primary key).

But I'm fine in keeping this check if you think it makes more sense.

Reply:
Makes sense. Good catch!